src/hooks/useDPContext.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from "react";
+import { useDeskproLatestAppContext } from "@deskpro/app-sdk";
+import { isUser, isOrganisation } from "@/utils";
+import type {
+  Maybe,
+  DPUser,
+  UserContext,
+  DPOrganisation,
+  OrganisationContext,
+} from "@/types";
+
+export type Result = {
+  isUserCtx: boolean;
+  isOrgCtx: boolean;
+  dpUser: Maybe<DPUser>;
+  dpOrg: Maybe<DPOrganisation>;
+};
+
+export type UseDPContext = () => Result;
+
+const useDPContext: UseDPContext = () => {
+  const { context } = useDeskproLatestAppContext() as { context?: UserContext|OrganisationContext };
+  const [isUserCtx, setIsUserCtx] = useState<boolean>(false);
+  const [isOrgCtx, setIsOrgCtx] = useState<boolean>(false);
+  const [dpUser, setDPUser] = useState<Maybe<DPUser>>(null);
+  const [dpOrg, setDPOrg] = useState<Maybe<DPOrganisation>>(null);
+
+  useEffect(() => {
+    if (isUser(context)) {
+      setIsUserCtx(true);
+      setDPUser(context?.data?.user);
```

Comment:
I removed one because otherwise, we have an error
<img width="475" alt="Screenshot 2024-08-02 at 16 34 50" src="https://github.com/user-attachments/assets/a2460544-6384-4b32-9791-d701f765fc9b">